### PR TITLE
Tests for specialized interface

### PR DESCRIPTION
**Pull request by abelsiqueira**
This implements the tests for the specialized interface. The same functions tested with the core interface are tested here.

**This was built on top of the PR #24.**

I think now I'm gonna try to implement the TRON method we discussed before.

### CODE REVIEW

File: src/core_interface.jl [hunk 1]
```diff
@@ -23,74 +24,81 @@ function csetup(io_err::Array{Cint, 1}, input::Array{Cint, 1}, out::Array{Cint,
...
+function udimen(io_err::Array{Cint, 1}, input::Array{Cint, 1},
+    n::Array{Cint, 1}, libname = fixedlibname)
   @eval ccall(("cutest_udimen_", $(libname)), Void,
     (Ptr{Cint}, Ptr{Cint}, Ptr{Cint}),
     $(io_err), $(input), $(n))
 end
 
-function udimsh(io_err::Array{Cint, 1}, nnzh::Array{Cint, 1}, libname = fixedlibname)
+function udimsh(io_err::Array{Cint, 1}, nnzh::Array{Cint, 1}, libname
+    = fixedlibname)
```
Comment (dpo) [inline src/core_interface.jl]: Did you mean to introduce a line break like this?

File: src/core_interface.jl [hunk 2]
```diff
@@ -23,74 +24,81 @@ function csetup(io_err::Array{Cint, 1}, input::Array{Cint, 1}, out::Array{Cint,
...
   @eval ccall(("cutest_udimsh_", $(libname)), Void,
     (Ptr{Cint}, Ptr{Cint}),
     $(io_err), $(nnzh))
 end
 
-function udimse(io_err::Array{Cint, 1}, ne::Array{Cint, 1}, he_val_ne::Array{Cint, 1},
-    he_row_ne::Array{Cint, 1}, libname = fixedlibname)
+function udimse(io_err::Array{Cint, 1}, ne::Array{Cint, 1},
+    he_val_ne::Array{Cint, 1}, he_row_ne::Array{Cint, 1}, libname =
+    fixedlibname)
```
Comment (dpo) [inline src/core_interface.jl]: Here's another one.

File: src/core_interface.jl [hunk 3]
```diff
@@ -504,13 +528,13 @@ end
...
   @eval ccall(("cutest_uterminate_", $(libname)), Void,
-    (Ptr{Cint},),
+    (Ptr{Cint}, ),
     $(io_err))
 end
 
 function cterminate(io_err::Array{Cint, 1}, libname = fixedlibname)
   @eval ccall(("cutest_cterminate_", $(libname)), Void,
-    (Ptr{Cint},),
+    (Ptr{Cint}, ),
```
Comment (dpo) [inline src/core_interface.jl]: All this reformatting is a bit strange, and I'm not sure it's necessary.

File: src/specialized_interface.jl [hunk 4]
```diff
@@ -0,0 +1,2725 @@
...
+    jl_cdimsh!, jl_cdimse!, jl_cstats!, jl_cvartype!, jl_cnames!,
+    jl_creport!, jl_connames!, jl_pname!, jl_probname!, jl_varnames!,
+    jl_ufn!, jl_ugr!, jl_uofg!, jl_ubandh!, jl_udh!, jl_ush!, jl_ueh!,
+    jl_ugrdh!, jl_ugrsh!, jl_ugreh!, jl_uhprod!, jl_cfn!, jl_cofg!,
+    jl_cofsg!, jl_ccfg!, jl_clfg!, jl_cgr!, jl_csgr!, jl_ccfsg!,
+    jl_ccifg!, jl_ccifsg!, jl_cgrdh!, jl_cdh!, jl_csh!, jl_cshc!, jl_ceh!,
+    jl_cidh!, jl_cish!, jl_csgrsh!, jl_csgreh!, jl_chprod!, jl_chcprod!,
+    jl_cjprod!, jl_uterminate!, jl_cterminate!
+
+function jl_usetup(input::Int, out::Int, io_buffer::Int, n::Int, libname = fixedlibname)
```
Comment (dpo) [inline src/specialized_interface.jl]: With multiple dispatch, couldn't all these functions have the same name and different signatures? Depending on the signature, `usetup()` could refer to the core or specialized interface, couldn't it? We just have to be a bit careful about the order of the arguments.

**Comment by dpo** [comment]
Should we close #24 then?

**Comment by abelsiqueira** [comment]
#24 closed.

File: src/core_interface.jl (hunk 1 above)
Comment (abelsiqueira) [inline src/core_interface.jl]: This, and others, happened because I changed some things in the script. In this case, the line wrapper. I'll fix it.

**Comment by abelsiqueira** [comment]
Commit 1b97091 removes the changes in the script. File `src/core_interface.jl` is unmodified.

**Comment by abelsiqueira** [comment]
About the multiple dispatch, I tested locally and it works. Also, we are currently not exporting the functions from core. `CUTEst.ufn` if from core and `ufn` is from the specialized. Should we export them too?

**Comment by dpo** [comment]
Yes I guess we should export all interfaces.

**Comment by abelsiqueira** [comment]
Updated the names and the export.

**Comment by abelsiqueira** [comment]
@dpo, do I need to rebase this, or you want to merge manually?

**Comment by dpo** [comment]
Sorry I haven't gotten to this yet. It would be great if you could rebase, if you don't mind. Thanks!

**Comment by abelsiqueira** [comment]
@dpo I refactored this PR into PR #33. It's easier to read, and cleaner. I'm closing this, but I'll wait until you merge #33 to delete this branch, in case you want to check individual commits.

**Pull request closed without merging**